### PR TITLE
Added tests for Elytron in Undertow subsystem

### PR DESCRIPTION
**Pull request by honza-kasik**
There is missing one test - referencing deployments, for which I need more time to investigate how to test it. This bunch of tests is ready to be merged!

### CODE REVIEW

File: common/src/main/java/org/jboss/hal/testsuite/page/config/ApplicationSecurityDomainPage.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.jboss.hal.testsuite.finder.Application;
+import org.jboss.hal.testsuite.finder.FinderNames;
+import org.jboss.hal.testsuite.fragment.ConfigAreaFragment;
+import org.jboss.hal.testsuite.fragment.ConfigFragment;
+import org.jboss.hal.testsuite.fragment.config.undertow.AddApplicationSecurityDomainWizard;
+import org.jboss.hal.testsuite.fragment.config.undertow.AddSSOWizard;
+import org.jboss.hal.testsuite.util.Console;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
```
Comment (zebrik) [inline common/src/main/java/org/jboss/hal/testsuite/page/config/ApplicationSecurityDomainPage.java:14]: Please add javadoc - at least that this is page for Undertow application security domain configuration (yes it's clear from the navigation method but should be in javadoc as well IMHO)

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/applicationsecuritydomain/ApplicationSecurityDomainOperations.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import org.jboss.hal.testsuite.test.configuration.elytron.ElytronOperations;
+import org.jboss.hal.testsuite.util.ConfigUtils;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+
+class ApplicationSecurityDomainOperations {
```
Comment (zebrik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/applicationsecuritydomain/ApplicationSecurityDomainOperations.java]: It seems that this class is completely replaced by `UndertowElytronOperations`. If this is true, please remove `ApplicationSecurityDomainOperations`.

**Review by zebrik — CHANGES_REQUESTED** [review]
Please remove duplicate `ApplicationSecurityDomainOperations` class.

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/applicationsecuritydomain/ApplicationSecurityDomainOperations.java (hunk 2 above)
Comment (honza-kasik) [reply basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/applicationsecuritydomain/ApplicationSecurityDomainOperations.java]: Sorry, copy paste issue, forget to delete

**Comment by honza-kasik** [comment]
Thanks for spotting duplicity errors and missing javadoc! I fixed all and squashed it!

**Review by zebrik — APPROVED** [review]

**Comment by zebrik** [comment]
LGTM. Just waiting for checkstyle results to merge. Thx!

**Pull request merged**